### ISSUE TITLE
[Miniflare] Emulate racing in KV

### ISSUE THREAD

**Issue by GregBrimble** [cloudflare/workers-sdk#4388]
Just an idea, and a relatively big project, so I have no idea if you'd want to support this in Miniflare, but might be interesting to have an option to emulate the quirks of racing KV writes.

Requests to a single colo can be raced when they hit a different machine, and of course, requests to different colos can be raced.

I imagine you'd want a something like 1/5 requests to 'hit a different machine'. This would be a good default behavior to allow people to set expectations of KV when they're prototyping something locally, or you could hide it away behind a flag if you don't want a breaking change.

TL;DR:

1. POST /increment -> 1 (machine 1)
1. POST /increment -> 2 (machine 1)
1. POST /increment -> 3 (machine 1)
1. POST /increment -> 1 (machine 2)
1. POST /increment -> 4 (machine 1)
1. POST /increment -> 5 (machine 1)
1. POST /increment -> 1 (machine 3)
1. POST /increment -> 2 (machine 2)

Expire each machine's cache after 60s and find the most recently written to value across the network.

**Comment by penalosa** [comment]
This is a very old issue so I'm going to close it for now. We may re-evaluate in future, but for now we're going to stick with local KV not being eventually consistent.
